belgif-rest-problem-java-ee/src/main/java/io/github/belgif/rest/problem/jaxrs/JaxRsParameterNameProvider.java

```diff
@@ -0,0 +1,70 @@
+package io.github.belgif.rest.problem.jaxrs;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.validation.ParameterNameProvider;
+import javax.ws.rs.CookieParam;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.MatrixParam;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * A ParameterNameProvider that retrieves the parameter name from JAX-RS annotations (if present).
+ *
+ * @see ParameterNameProvider
+ */
+@Provider
+public class JaxRsParameterNameProvider implements ParameterNameProvider {
+
+    private final ConcurrentHashMap<Executable, List<String>> parameterNameCache = new ConcurrentHashMap<>();
```

Comment:
Maybe we should make this cache static? Because I don't know if the JAX-RS / Bean Validation spec defines that there will only be a single reused instance of the ParameterNameProvider.

Reply:
The interface documentation mentions that the implementation should be thread-safe, so I assume it would be reused? It wouldn't hurt to make it static though, that way we're certain.